### DISCUSSION TITLE
Edge runtime for middle ware

### DISCUSSION THREAD

**Discussion by mohitsamant2k** [vercel/next.js#70132] [Help] [answered]
### Summary

If I deploy my next js app in cloud provider that do not support edge computing, then in which runtime my middleware will run in next js app? WIll the middleware run in server and but in edge runtime?

**Comment by leerob** [comment, marked as answer]
Good news everyone, **Node.js Middleware support landed in Next.js 15.2** (experimental). This is now ready for testing. For any issues you run into, please open new issues so we can track and address before marking it as stable. Thank you!

Notably, Middleware is still supported when self-hosting, regardless of if you use the edge runtime or not.

https://nextjs.org/blog/next-15-2#nodejs-middleware-experimental
